source/common/ur_params.hpp

```diff
@@ -0,0 +1,7780 @@
+        os << "UR_KERNEL_SUB_GROUP_INFO_MAX_NUM_SUB_GROUPS";
+        break;
+
+    case UR_KERNEL_SUB_GROUP_INFO_COMPILE_NUM_SUB_GROUPS:
+        os << "UR_KERNEL_SUB_GROUP_INFO_COMPILE_NUM_SUB_GROUPS";
+        break;
+
+    case UR_KERNEL_SUB_GROUP_INFO_SUB_GROUP_SIZE_INTEL:
+        os << "UR_KERNEL_SUB_GROUP_INFO_SUB_GROUP_SIZE_INTEL";
+        break;
+    default:
+        os << (value == 0 ? "0" : "invalid enumerator");
+        break;
+    }
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os,
+                                enum ur_kernel_exec_info_t value) {
+    switch (value) {
+
+    case UR_KERNEL_EXEC_INFO_USM_INDIRECT_ACCESS:
+        os << "UR_KERNEL_EXEC_INFO_USM_INDIRECT_ACCESS";
+        break;
+
+    case UR_KERNEL_EXEC_INFO_USM_PTRS:
+        os << "UR_KERNEL_EXEC_INFO_USM_PTRS";
+        break;
+    default:
+        os << (value == 0 ? "0" : "invalid enumerator");
+        break;
+    }
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os, enum ur_queue_info_t value) {
+    switch (value) {
+
+    case UR_QUEUE_INFO_CONTEXT:
+        os << "UR_QUEUE_INFO_CONTEXT";
+        break;
+
+    case UR_QUEUE_INFO_DEVICE:
+        os << "UR_QUEUE_INFO_DEVICE";
+        break;
+
+    case UR_QUEUE_INFO_DEVICE_DEFAULT:
+        os << "UR_QUEUE_INFO_DEVICE_DEFAULT";
+        break;
+
+    case UR_QUEUE_INFO_PROPERTIES:
+        os << "UR_QUEUE_INFO_PROPERTIES";
+        break;
+
+    case UR_QUEUE_INFO_REFERENCE_COUNT:
+        os << "UR_QUEUE_INFO_REFERENCE_COUNT";
+        break;
+
+    case UR_QUEUE_INFO_SIZE:
+        os << "UR_QUEUE_INFO_SIZE";
+        break;
+    default:
+        os << (value == 0 ? "0" : "invalid enumerator");
+        break;
+    }
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os, enum ur_queue_flag_t value) {
+    switch (value) {
+
+    case UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE:
+        os << "UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE";
+        break;
+
+    case UR_QUEUE_FLAG_PROFILING_ENABLE:
+        os << "UR_QUEUE_FLAG_PROFILING_ENABLE";
+        break;
+
+    case UR_QUEUE_FLAG_ON_DEVICE:
+        os << "UR_QUEUE_FLAG_ON_DEVICE";
+        break;
+
+    case UR_QUEUE_FLAG_ON_DEVICE_DEFAULT:
+        os << "UR_QUEUE_FLAG_ON_DEVICE_DEFAULT";
+        break;
+
+    case UR_QUEUE_FLAG_DISCARD_EVENTS:
+        os << "UR_QUEUE_FLAG_DISCARD_EVENTS";
+        break;
+
+    case UR_QUEUE_FLAG_PRIORITY_LOW:
+        os << "UR_QUEUE_FLAG_PRIORITY_LOW";
+        break;
+
+    case UR_QUEUE_FLAG_PRIORITY_HIGH:
+        os << "UR_QUEUE_FLAG_PRIORITY_HIGH";
+        break;
+    default:
+        os << (value == 0 ? "0" : "invalid enumerator");
+        break;
+    }
+    return os;
+}
+inline void serializeFlag_ur_queue_flags_t(std::ostream &os,
+                                           ur_queue_flags_t flag) {
+    uint32_t val = flag;
+    bool first = true;
+
+    if ((val & UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE) ==
```

Comment:
can this flag cehcking code be made generic? It's seems like a simple loop should be enough

Reply:
Over bits? But that would require us to cast it back to the appropriate enum type if we want to actually print useful information. And how do you know when to stop the loop? Would we iterate over all the bits in a type?

I don't know. That doesn't seem simpler to me.

Reply:
I mean you could just generate a list of values in mako using `%for n, item in enumerate(obj['etors']):` and then just have a for loop here over those values, but if it's easier to implement in mako this way then it's fine.

Reply:
so something like:
```
parseFlagsGeneric<T>(T flag, T all_flags[]) {
   for (f : flags) { 
     if ((flag & f) == f) os << f;
   }
}

parseFlags_foo(enum foo_flag flag) {
   enum foo_flag all_etors[] = {FOO_A, FOO_B};
   parseFlagsGeneric(flag, all_etors);
}
```
(or without the template function)

I guess either method will work, I don't think one is better than the other. It's all autogenerated anyway.